task_3.js

```diff
@@ -0,0 +1,118 @@
+var express = require('express');
+var rsvp    = require('rsvp');
+var app     = express();
+var http    = require('http')
+
+var PATH = '/I/want/title';
+
+app.get(PATH, function (req, res) {
+
+  addresses = parseURL(req.url);
+  if(addresses){
+    titles  = [];
+    list_of_titles = '';
+
+    var promise = new rsvp.Promise(function(resolve, reject) {
+      getTitles(addresses, resolve);
```

Comment:
What if we need to reject in case of error, we are not passing reject to method, so it can't reject the promise chain.